mods/sfinv/api.lua

```diff
@@ -0,0 +1,167 @@
+sfinv = {
+	pages = {},
+	pages_unordered = {},
+	homepage_name = "sfinv:crafting",
+	contexts = {},
+	enabled = true
+}
+
+function sfinv.register_page(name, def)
+	if not name or not def or not def.get then
```

Comment:
On the other hand, `error()` allows the stack level to be defined as additional parameter, making it clear, where the error really happens.

```lua
error("Invalid sfinv page. Requires a name & def, and a get function in def", 2)
```